src/java/voldemort/utils/StoreDefinitionUtils.java

```diff
@@ -254,6 +255,81 @@ public static void validateSchemasAsNeeded(Collection<StoreDefinition> storeDefi
         }
     }
 
+    /**
+     * Ensure that new store definitions that are specified for an update do not include breaking changes to the store.
+     * @param oldStoreDefs
+     * @param newStoreDefs
+     */
+    public static void validateNewStoreDefsAreNonBreaking(List<StoreDefinition> oldStoreDefs, List<StoreDefinition> newStoreDefs){
+        Map<String, StoreDefinition> oldStoreMap = new HashMap<String, StoreDefinition>();
+        Map<String, StoreDefinition> newStoreMap = new HashMap<String, StoreDefinition>();
+        for (StoreDefinition storeDef : oldStoreDefs){
+           oldStoreMap.put(storeDef.getName(), storeDef);
+        }
+        for (StoreDefinition storeDef : newStoreDefs){
+           newStoreMap.put(storeDef.getName(), storeDef);
+        }
+        for (String storeName : oldStoreMap.keySet()){
+            if (newStoreMap.containsKey(storeName)){
+               validateNewStoreDefIsNonBreaking(oldStoreMap.get(storeName), newStoreMap.get(storeName));
+            }
+        }
+    }
+
+    /**
+     * Ensure that new store definitions that are specified for an update do not include breaking changes to the store.
+     *
+     * Non-breaking changes include changes to
+     *   description
+     *   replicationFactor
+     *   preferredWrites
+     *   requiredWrites
+     *   preferredReads
+     *   requiredReads
+     *   retentionPeriodDays
+     *   retentionScanThrottleRate
+     *   retentionFrequencyDays -
+     *   zoneCountReads
+     *   zoneCountWrites
+     *   owners
+     *   memoryFootprintMB
+     *
+     * @param oldStoreDef
+     * @param newStoreDef
+     */
+    public static void validateNewStoreDefIsNonBreaking(StoreDefinition oldStoreDef, StoreDefinition newStoreDef){
+        if (!oldStoreDef.getName().equals(newStoreDef.getName())){
+            throw new VoldemortException("Cannot compare stores of different names: " + oldStoreDef.getName() + " and " + newStoreDef.getName());
+        }
+      String store = oldStoreDef.getName();
+        verifySamePropertyForUpdate(oldStoreDef.getType(), newStoreDef.getType(), "Type", store);
+        verifySamePropertyForUpdate(oldStoreDef.getKeySerializer(), newStoreDef.getKeySerializer(), "KeySerializer", store);
```

Comment:
In RW, key serializer, value serializer and transform serializer can be evolved.